### PR TITLE
RTA: get amount from transaction 

### PR DESCRIPTION
**Pull request by mbg033**
- RTA related funcitonality: check if any outputs in pending transaction object addressed to particular wallet

### CODE REVIEW

File: src/supernode/PosSaleObject.cpp [hunk 1]
```diff
@@ -43,65 +44,8 @@ namespace  {
 
 // TODO: move these functions to some "Utils" class/library
```
[the remark this answers is not in the document]
Comment (mbg033) [reply src/supernode/PosSaleObject.cpp]: fixed

File: src/utils/utils.cpp [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+namespace Utils {
+
+bool decode_ringct(const rct::rctSig& rv, const crypto::public_key pub, const crypto::secret_key &sec, unsigned int i, rct::key & mask, uint64_t & amount)
+{
+    crypto::key_derivation derivation;
+    bool r = crypto::generate_key_derivation(pub, sec, derivation);
+    if (!r)
+    {
+        LOG_ERROR("Failed to generate key derivation to decode rct output " << i);
+        return 0;
```
[the remark this answers is not in the document]
Comment (mbg033) [reply src/utils/utils.cpp]: fixed

File: src/supernode/graft_wallet.cpp [hunk 3]
```diff
@@ -4195,6 +4196,14 @@ bool GraftWallet::load_keys_graft(const string &data, const string &password)
     return true;
 }
 
+bool GraftWallet::get_amount_from_tx(const GraftWallet::pending_tx &ptx, uint64_t &amount)
+{
+  std::vector<std::pair<size_t, uint64_t>> unused;
+  bool r = ::Utils::lookup_account_outputs_ringct(this->m_account.get_keys(), ptx.tx, unused, amount);
+  return r;
```
[the remark this answers is not in the document]
Comment (mbg033) [reply src/supernode/graft_wallet.cpp]: removed extra variable

File: src/wallet/wallet2.cpp [hunk 4]
```diff
@@ -5967,4 +5969,15 @@ void wallet2::generate_genesis(cryptonote::block& b) {
     cryptonote::generate_genesis_block(b, config::GENESIS_TX, config::GENESIS_NONCE);
   }
 }
+//----------------------------------------------------------------------------------------------------
+bool wallet2::get_amount_from_tx(const wallet2::pending_tx &ptx, uint64_t &amount)
+{
+  std::vector<std::pair<size_t, uint64_t>> unused;
+  bool r = Utils::lookup_account_outputs_ringct(this->m_account.get_keys(), ptx.tx, unused, amount);
```
Comment (EDDragonWolf) [inline src/wallet/wallet2.cpp]: Also here

Comment (mbg033) [reply src/wallet/wallet2.cpp]: removed one more extra variable

**Pull request merged**